theme/scoreboard.html

```diff
@@ -4,130 +4,14 @@
-  let pointshistory = JSON.parse(localStorage.getItem("pointshistory")) || [];
-  if (pointshistory.length >= 20){
-    pointshistory.shift();
-  }
-  pointshistory.push(pointslog);
-  localStorage.setItem("pointshistory", JSON.stringify(pointshistory));
-
-  // Dole out points
-  for (let i in pointslog) {
-    let entry = pointslog[i];
-    let timestamp = entry[0];
-    let teamhash = entry[1];
-    let category = entry[2];
-    let points = entry[3];
-
-    let team = teams[teamhash] || {__hash__: teamhash};
-
-    // Add points to team's points for that category
-    team[category] = (team[category] || 0) + points;
-
-    // Record highest score in a category
-    highscore[category] = Math.max(highscore[category] || 0, team[category]);
-
-    teams[teamhash] = team;
-  }
-
-  // Sort by team score
-  function teamScore(t) {
-    let score = 0;
-    for (let category in highscore) {
-      score += (t[category] || 0) / highscore[category];
-    }
-    return score;
-  }
-  function teamCompare(a, b) {
-    return teamScore(a) - teamScore(b);
-  }
-
-  // Figure out how to order each team on the scoreboard
-  let winners = [];
-  for (let i in teams) {
-    winners.push(teams[i]);
-  }
-  winners.sort(teamCompare);
-  winners.reverse();
-
-  // Clear out the element we're about to populate
-  Array.from(element.childNodes).map(e => e.remove());
-
-  let maxWidth = 100 / Object.keys(highscore).length;
-  for (let i in winners) {
-    let team = winners[i];
-    let row = document.createElement("div");
-    let ncat = 0;
-    for (let category in highscore) {
-      let catHigh = highscore[category];
-      let catTeam = team[category] || 0;
-      let catPct = catTeam / catHigh;
-      let width = maxWidth * catPct;
-
-      let bar = document.createElement("span");
-      bar.classList.add("category");
-      bar.classList.add("cat" + ncat);
-      bar.style.width = width + "%";
-      bar.textContent = category + ": " + catTeam;
-      bar.title = bar.textContent;
-
-      row.appendChild(bar);
-      ncat += 1;
-    }
-
-    let te = document.createElement("span");
-    te.classList.add("teamname");
-    te.textContent = teamnames[team.__hash__];
-    row.appendChild(te);
-
-    element.appendChild(row);
-  }
-}
-
-function once() {
-  fetch("points.json")
-  .then(resp => {
-    return resp.json();
-  })
-  .then(obj => {
-    update(obj);
-  })
-  .catch(err => {
-    console.log(err);
-  });
-}
-
-function init() {
-  let base = window.location.href.replace("scoreboard.html", "")
-  document.querySelector("#location").textContent = base
-
-  setInterval(once, 60000);
-  once();
-}
-if (document.readyState === "loading") {
-  document.addEventListener("DOMContentLoaded", init);
-} else {
-  init();
-}
-    </script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.24.0/moment.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/chart.js@2.8.0/dist/Chart.min.js" async></script>
```

Comment:
Are you sure about adding external dependencies? This may break some PWA caching, but we don't care too much about the scoreboard.

Reply:
Good catch, I'll add the file to our theme directory.